### PR TITLE
soc: arm: silabs_exx32: DT_GPIO_GECKO_SWO_LOCATION rework

### PR DESCRIPTION
**Pull request by galak**
Change DT_GPIO_GECKO_SWO_LOCATION to use new DT_INST and DT_PROP macros
and remove defining it in dts_fixup.h.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

File: soc/arm/silabs_exx32/efm32gg11b/soc_pinmap.h [hunk 1]
```diff
@@ -18,17 +18,22 @@
 #include <soc.h>
 #include <em_gpio.h>
 
+#define GPIO_NODE DT_INST(0, silabs_gecko_gpio)
+#if DT_NODE_HAS_PROP(GPIO_NODE, swo_location)
```
Comment (chrta) [inline soc/arm/silabs_exx32/efm32gg11b/soc_pinmap.h:22]: swo_location -> location_swo (also in the next line)

**Review by chrta — CHANGES_REQUESTED** [review]
I tested this on the efm32pg board and after the mentioned change it worked fine.

File: soc/arm/silabs_exx32/efm32gg11b/soc_pinmap.h (hunk 1 above)
Comment (galak) [reply soc/arm/silabs_exx32/efm32gg11b/soc_pinmap.h]: Thanks for testing, should now be fixed.

**Review by chrta — APPROVED** [review]
Works fine on efm32pg board

**Comment by mnkp** [comment]
The code looks good, however the SWO code was written before we figured out how to handle location / pinmuxing in the gecko family. To keep it aligned with other peripherals we should do it as
```
&gpio {
	location-swo = <GECKO_LOCATION(0) GECKO_PORT_F GECKO_PIN(2)>;
	status = "okay";
};
```
This way we could get rid also of `soc_pinmap.h` files. I can prepare a PR, however that would be a breaking change. I mean, out of tree code would be affected.

We can also do it later, after merging this PR.

**Comment by galak** [comment]
> The code looks good, however the SWO code was written before we figured out how to handle location / pinmuxing in the gecko family. To keep it aligned with other peripherals we should do it as
> 
> ```
> &gpio {
> 	location-swo = <GECKO_LOCATION(0) GECKO_PORT_F GECKO_PIN(2)>;
> 	status = "okay";
> };
> ```
> 
> This way we could get rid also of `soc_pinmap.h` files. I can prepare a PR, however that would be a breaking change. I mean, out of tree code would be affected.
> 
> We can also do it later, after merging this PR.

(But do you approve of this PR ;).

**Review by mnkp — APPROVED** [review]

**Pull request merged**